addons/survey/static/src/scss/survey_survey_views.scss

```
@@ -142,6 +142,17 @@
         background: transparent;
     }
 }
+.o_survey_form table.o_section_list_view tr.o_data_row.o_is_section:hover {
+    i{
+        visibility: visible;
+    }
+}
+
+.o_survey_form table.o_section_list_view tr.o_data_row.o_is_section:not(:hover) {
+    i{
+        visibility: hidden;
+    }
+}
```

You can get a few less lines with
```suggestion
.o_survey_form table.o_section_list_view tr.o_data_row.o_is_section {
    &:hover i {
        visibility: visible;
    }
    &:not(hover) i {
        visibility: invisible;
    }
}
```